src/Messages/OWSMessageSender.m

```diff
@@ -554,6 +560,9 @@ - (void)handleMessageSentLocally:(TSOutgoingMessage *)message
 {
     [self saveMessage:message withState:TSOutgoingMessageStateSent];
     if (message.shouldSyncTranscript) {
+        // TODO: I suspect we shouldn't optimistically set hasSyncedTranscript.
+        //       We could set this in a success handler for [sendSyncTranscriptForMessage:].
```

Comment:
I've opened a new issue where we can track/discuss this issue: https://github.com/WhisperSystems/Signal-iOS/issues/1761